deps.ffmpeg/60-mbedtls.zsh

```diff
@@ -73,22 +73,24 @@ config() {
     -DGEN_FILES=OFF
   )
 
+  if [[ ${config} == Release ]] args+=(-DCMAKE_C_FLAGS="${c_flags} -std=c17 -g")
```

Comment:
Do we only set c17 in Release?

Reply:
No, but the "global" `CMAKE_C_FLAGS` variable is defined as `"${c_flags} -std=c17"` and if we want to append to that (as we need to do for Release to ensure dSYMs are generated), we need to replace the entire value on the command line (and thus repeat it here).

Reply:
`CMAKE_C_FLAGS` are usually empty, instead the configuration-specific flags are set.

CMake itself adds `-g` (which enables debug information) for Debug and RelWithDebInfo only.

On macOS it is canonical to _always_ have debug symbols, but embed them in Debug and RelWithDebInfo (`DWARF`) and compile them into separate debug symbols (`DWARF with dSYM`) otherwise. Alas CMake does not support this behaviour.

Instead the `-g` flag needs to be added to compilation flags (as CMake itself checks for its presence in the flags to enable debug symbol generation). This happens inside CMake's generators, so generator-specific options are ignored.

In `30-build.zsh` we do just that: `-DCMAKE_C_FLAGS_RELEASE="-O3 -g -DNDEBUG"`.

MbedTLS however sets/overwrites `CMAKE_C_FLAGS_RELEASE`, so we lose this information. And the only way to get the `-g` flag back is to add it to the "global" `CMAKE_C_FLAGS` (which are empty in MbedTLS' case) for the release config only.

Reply:
I think I understand.